### PR TITLE
Import VPack Deserializer

### PR DESCRIPTION
**Pull request by markuspf**
This imports @maierlars template velocypack deserialization library. 

It allows for defining parsers for VelocyPack slices, making sure that requried fields are present and of the correct type, and storing data into a struct.

TODO:

 - [x] Disclaimers in every file
 - [x] Some simple examples to get started
 - [x] Tests
 - [x] More detailed documentation.
 - [x] Tests should really be run as part of testing arangodb.

### CODE REVIEW

File: lib/VPackDeserialize/values.h [hunk 1]
```diff
@@ -129,8 +129,7 @@ struct deserialize_plan_executor<values::value_deserializer<T>, H> {
 
   template<typename C>
   static auto unpack(::deserializer::slice_type s, typename H::state_type hints, C&&) -> result_type {
-    ensure_value_reader<T> guard;
-    (void)guard;
```
Comment (markuspf) [inline lib/VPackDeserialize/values.h]: Will this change break compilation for me again?

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/11072/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/11079/

File: lib/VPackDeserializer/fixed-order.h [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+}  // namespace detail
+
+template <typename... Ds, typename H>
+struct deserialize_plan_executor<fixed_order_deserializer<Ds...>, H> {
+  using value_type = typename fixed_order_deserializer<Ds...>::constructed_type;
+  using tuple_type = value_type;  // std::tuple<value_type>;
+  using result_type = result<tuple_type, deserialize_error>;
+
+  constexpr static auto expected_array_length = sizeof...(Ds);
```
Comment (mchacki) [inline lib/VPackDeserializer/fixed-order.h:103]: If the response to my following comment is "not needed, can be removed"
Then this value is of no need anymore, and can be removed as well.

File: lib/VPackDeserializer/fixed-order.h [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    if (!s.isArray()) {
+      return result_type{deserialize_error{"expected array"}};
+    }
+
+    /*if (s.length() != expected_array_length) {
+      return result_type{deserialize_error{
+          "bad array length, found: "s + std::to_string(s.length()) +
+          ", expected: " + std::to_string(expected_array_length)}};
+    }*/
```
Comment (markuspf) [inline lib/VPackDeserializer/fixed-order.h:118]: Is this needed? Delete or use.

File: lib/VPackDeserializer/fixed-order.h [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                            std::to_string(expected_array_length)};
+    }
+
+    /*
+        bool result =
+            (deserialize_with<Ds>(*iter++).visit(
+                 detail::fixed_order_deserializer_executor_visitor<I, typename
+       Ds::constructed_type, deserialize_error>( std::get<I>(values), error)) &&
+             ...);
+    */
```
Comment (markuspf) [inline lib/VPackDeserializer/fixed-order.h:166]: Is this needed?

File: lib/VPackDeserializer/values.h [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+struct is_string<string_value<V>> : std::true_type {};
+template <typename V>
+constexpr bool is_string_v = is_string<V>::value;
+
+template <typename T>
+struct default_constructed_value {
+  constexpr static T value = T{};
+};
+
+// TODO add proper static_asserts to ensure the correct behavior for values.
```
Comment (mchacki) [inline lib/VPackDeserializer/values.h:71]: Is this still valid?

File: lib/VPackDeserializer/unpack-proxy.h [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Lars Maier
+///
+////////////////////////////////////////////////////////////////////////////////
+//
+// Created by lars on 30.12.19.
+//
```
Comment (markuspf) [inline lib/VPackDeserializer/unpack-proxy.h:26]: Probably good to delete this too ;)

File: lib/VPackDeserializer/values.h [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+// This should include:
+//  - having a `value` member.
+//  - having a `type` member type.
+//  - having a `to_string` implementation.
```
Comment (markuspf) [inline lib/VPackDeserializer/values.h:75]: Should this stay a TODO?

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/11086/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/11094/

**Comment by mchacki** [comment]
Jenkins after devel merge:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/11163

all blue

**Pull request merged**